core/domain/user_services.py

```diff
@@ -495,6 +495,96 @@ def get_user_settings_by_auth_id(
         return None
 
 
+def get_all_user_groups() -> List[Dict[str, Union[str, List[str]]]]:
+    """Return the list of user groups dictionary data having key as user-group
+    name and value as list of users associated to the user-group.
+
+    Returns:
+        user_group_models_dict: List[Dict[str, Union[str, List[str]]]]. List of
+        all user groups dictionary with key as user-group name and value as
+        list of users associated to user-group.
+    """
+    user_group_models: List[user_models.UserGroupModel] = list(
+        user_models.UserGroupModel.get_all())
+    user_group_models_list = []
+    for user_group_model in user_group_models:
+        user_group_models_dict: Dict[str, Union[str, List[str]]] = {}
+        user_group_models_dict['user_group_name'] = user_group_model.id
+        user_group_models_dict['users'] = list(
+            user_group_model.users)
+        user_group_models_list.append(user_group_models_dict)
+    return user_group_models_list
+
+
+def delete_user_group(user_group_to_delete: str) -> None:
+    """Delete the specified user group.
+
+    Args:
+        user_group_to_delete: str. The name of the user group that need to
+            be removed.
+
+    Raises:
+        Exception. The user group trying to delete does not exists.
+    """
+    user_group_model = user_models.UserGroupModel.get(
+        user_group_to_delete, strict=False)
+    if user_group_model is None:
+        raise Exception(f'User group {user_group_to_delete} does not exist.')
+    assert user_group_model is not None
+    user_group_model.delete()
+
+
+def update_user_group(
```

Comment:
I don't think you're modelling this correctly. I think what you need is an immutable ID for each user group (set upon creation), and a human-readable name which can be changed.

This is because, when you specify a user group in the config for a feature flag, you'll need to do that specification by ID. If you do it by a mutable name, then if that name changes, the feature flags will point to an incorrect reference.

Reply:
Got it, I have updated this.